MM-60224 - disable upgrade app button once download starts

pvev opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
There was a problem with the upgrade app button were once the user clicked it and the download started, the button kept enabled so users could click it multiple times causing the app to crash, even thought the download and the upgrade completed successfully. This PR makes sure to disable the button once the download has started and provides feedback using a loading spinner and showing Downloading text in the button.

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/desktop/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
https://mattermost.atlassian.net/browse/MM-60224
#### Checklist
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
- [ ] completed [Mattermost Contributor Agreement](https://mattermost.com/contribute/)
- [ ] executed `npm run lint:js` for proper code formatting
- [ ] Run E2E tests by adding label `Run Desktop E2E Tests`

#### Device Information
This PR was tested on: <!-- Device name(s), OS version(s) -->

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->
Before:
https://github.com/user-attachments/assets/96ad3122-44be-4d98-b3e0-780529c4fb93

After:
https://github.com/user-attachments/assets/300c6f22-eb15-42e4-adc4-be7f1a0ec5d7

#### Release Note
<!--
Add a release note for each of the following conditions:

* New features and improvements, including behavioural changes, UI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense. Newlines are stripped.

Examples:

```release-note
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->

```release-note
NONE

```

devinbinnie reviewed (approved):
Perfect :)

jespino reviewed (approved):
LGTM

yasserfaraazkhan commented:
Hi @devinbinnie @pvev 
how do we reduce the version and test this change ? can you share QA steps?

pvev commented:
> Hi @devinbinnie @pvev how do we reduce the version and test this change ? can you share QA steps?

hey @yasserfaraazkhan , sorry, I forgot to mentioned it here but I already asked @devinbinnie and I created a fake PR with 5.8 base + this changes to get the CI build the packages that will detect the upgrade to 5.9. PR is here: https://github.com/mattermost/desktop/pull/3169

yasserfaraazkhan reviewed (changes requested):
@pvev tested this on windows and macos
we can see the button get disabled.

I observed that in Dark more (OS's set it dark mode)
<img width="306" alt="Screenshot 2024-10-18 at 2 15 20 AM" src="https://github.com/user-attachments/assets/271a4dca-c9cb-4b00-86af-6d3dca7e7557">
 we dont see download button. 
 
 If this is an existing issue then its not a blocker. If not should this be fixed in this PR or separately. Please let me know.
 
<img width="313" alt="Screenshot 2024-10-18 at 2 15 15 AM" src="https://github.com/user-attachments/assets/428c7b63-d23e-493d-b262-8a4706392b18">

pvev commented:
> @pvev tested this on windows and macos we can see the button get disabled.
> 
> I observed that in Dark more (OS's set it dark mode) <img alt="Screenshot 2024-10-18 at 2 15 20 AM" width="306" src="https://private-user-images.githubusercontent.com/15979783/377629484-271a4dca-c9cb-4b00-86af-6d3dca7e7557.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0MTkyMDAsIm5iZiI6MTcyOTQxODkwMCwicGF0aCI6Ii8xNTk3OTc4My8zNzc2Mjk0ODQtMjcxYTRkY2EtYzljYi00YjAwLTg2YWYtNmQzZGNhN2U3NTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIwVDEwMDgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3NGViZWJmOTA3MjkxYWIwZmI2MjQ5MTdmZjlhMGNjNmUwYmMyMDE1OTdmYzUxZjgzMzY2NjIwNjA1YjRhNjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YzwzehtUJKO0wxvjikACSd1xR-dq8-GRSBWnwoQ8_R0"> we dont see download button.
> 
> If this is an existing issue then its not a blocker. If not should this be fixed in this PR or separately. Please let me know.
> 
> <img alt="Screenshot 2024-10-18 at 2 15 15 AM" width="313" src="https://private-user-images.githubusercontent.com/15979783/377629478-428c7b63-d23e-493d-b262-8a4706392b18.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0MTkyMDAsIm5iZiI6MTcyOTQxODkwMCwicGF0aCI6Ii8xNTk3OTc4My8zNzc2Mjk0NzgtNDI4YzdiNjMtZDIzZS00OTNkLWIyNjItOGE0NzA2MzkyYjE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIwVDEwMDgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjOTcyNDg2N2NkNDdlMTRlZjAxNjQ0MzY3NDAzNzNiMDkzYzY3MjNjYzUxYjA4Y2Q5NDBhOTY3YTczNWRmMjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4GwXS5YF9R4v4aqDDMl00Njz-JCtAbO_h7E0Peronx8">

fixed (and also fixed the clear all disabled button styles):

https://github.com/user-attachments/assets/cbacedcc-307f-42ee-9be9-9e459d4c79e2
